### ISSUE TITLE
Requests are not logged via Arcus.Observability in isolated Azure Functions with Http trigger

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.templates#785]
**Describe the bug**

When you create an Azure HTTP triggered function using the Arcus templates (v1.1.0), with Serilog logging configured to log to an Application Insights sinks, we can see that requests are being logged in AppInsights.
However, it seems that those requests are not being logged via the Arcus middleware, but this are rather logged via the Microsoft middleware.

This means that the requests that are being logged, are not as expected.  (f.i. the request body or response is not available although it has been configured, etc...)

The requests that are being logged, are logged because of the `APPLICATIONINSIGHTS_CONNECTION_STRING` configuration setting that is present in our config setting.  If I remove or rename this setting , no requests are being logged anymore.  This can be logical, as Azure Functions is actually looking for this setting to setup AppInsights logging.

I can see however that the warning traces that represent our 'request logging', are being written to Application Insights.  However, they're not converted to the 'request' format, they're rather written as a regular trace (with severity level warning):

![arcus-http-request](https://user-images.githubusercontent.com/3605786/218767524-a513064d-1ccf-4ff1-b5aa-e3b19a0ca0bb.png)

![arcus-http-request-warning-trace](https://user-images.githubusercontent.com/3605786/218767594-4549e2ad-c9e9-47c4-86e5-76011c2ae527.png)

Because of this, additional logs / traces that we write inside the Function are also not present when looking at the 'end-to-end' transaction details of a request.  

Consider this sample code:

![image](https://user-images.githubusercontent.com/3605786/218768712-b57af440-9203-4ca9-b4ee-16533c23f63a.png)

When looking in Application Insights at the 'performance' blade, we can see this:

![image](https://user-images.githubusercontent.com/3605786/218769010-50bd4355-8a9d-49c3-b8d2-5a70b28afa5f.png)

You can see that we see invocations for operation-name 'order' (which is the Function Name specified in the `Function` attribute). 
However, the requests that are logged here, are the requests that have been logged via the MSFT middleware (picked up because of the configured `APPLICATIONINSIGHTS_CONNECTION_STRING` setting).
When drilling into one of those invocations, we notice that the custom log-traces and dependencies are not present:

![image](https://user-images.githubusercontent.com/3605786/218769419-caa73733-2a84-4944-a9e6-c15d257e0de3.png)

We can find those specific traces though by using the 'Transaction Search' functionality in Application Insights:
![image](https://user-images.githubusercontent.com/3605786/218769706-1136fb27-0fba-42fb-aa05-a0a4e3867ef4.png)

![image](https://user-images.githubusercontent.com/3605786/218769816-6ffd6e45-d578-46ca-883f-1d7ef4b5987f.png)

Going into the details of these traces, we also notice that these traces do not have a parentId:

![image](https://user-images.githubusercontent.com/3605786/218770187-608b90d2-9da2-4ee0-9191-7a510df0955a.png)

**To Reproduce**

Create a new Http Triggered (isolated) Function project using the Arcus template.  Configure the Application Insights connection-string, run the project and perform some http calls.

```
dotnet new arcus-az-func-http -fw isolated --name sometestfunction
```

To be able to run this locally, make sure to correctly specify the `AzureWebJobsStorage` config setting in host.json:

```json
{
    "IsEncrypted": false,
  "Values": {
    "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated",
    "APPLICATIONINSIGHTS_CONNECTION_STRING": "<your-connectionstring>",
    "AzureWebJobsStorage": "UseDevelopmentStorage=true"
  }
}
```

**Comment by stijnmoreels** [comment]
Ah, I remember.
This is because of the tedious way that Azure Functions is set up. Because several internal workings are hidden and not available for customization, we had to be creative. Because we support W3C, we want the default Application Insights functionality combined with our easy-to-use logger infrastructure. To do that in Azure Functions, you need to provide the `APPLICATIONINSIGHTS_CONNECTION_STRING`. This will make sure that any additional stuff people want to write with the `TelemetryClient` will also be correlated like when they solely use Application Insights services.
The problem though, is that we can't disable request tracking when we use this way of configuring Application Insights, so, the way it is set up is that Arcus fully uses the default Application Insights request tracking in those cases. The fact that there is a middleware component for isolated Azure Functions is probably a mistake that we overlooked.

So, in a way, this is not a bug, but expected behavior. As commented in the referenced issue about unwanted traces, here's the things we should consider doing:
* Add `logging:logLevel:Host.Function=None` to `host.json`
* Deprecated `AzureFunctionsRequestTrackingMiddleware`

**Comment by fgheysels** [comment]
But what about correlating your own custom traces to the correct request, because they're currently not correlated.
Also, requests that are logged via the default Azure Functions middleware do not respect the role-name / app-name that we configure.

**Comment by stijnmoreels** [comment]
> But what about correlating your own custom traces to the correct request, because they're currently not correlated.

This should work already. Are you sure you selected the 'correct' traces? And not the unwanted traces from the host logger? Bc, if you select the correct ones, you see that they are correlated to the request and other stuff.

> Also, requests that are logged via the default Azure Functions middleware do not respect the role-name / app-name that we configure.

This is indeed something wrong with Azure Functions as they do not pick up `ITelemetryInitializer` instances, as they do not allow such customization. I imagine there may be some way, but haven't looked yet; sole focus was W3C service-to-service correlation; which finally works on all platforms.

**Comment by fgheysels** [comment]
> > But what about correlating your own custom traces to the correct request, because they're currently not correlated.
> 
> This should work already. Are you sure you selected the 'correct' traces? And not the unwanted traces from the host logger? Bc, if you select the correct ones, you see that they are correlated to the request and other stuff.

Yes.  Consider an Azure Function project, created via the Arcus template.  The only thing I've customized up until now, is just adding some additional logging.  
For example:

![image](https://user-images.githubusercontent.com/3605786/219370129-44640f76-3be5-4b9b-af43-05294d5222fc.png)

When opening Application Insights after a few minutes, I can see only one request in the last 24 hours (logical, since I haven't run the function since today):

![image](https://user-images.githubusercontent.com/3605786/219370391-a13da30c-f14d-467b-937e-d57f6c159f32.png)

Drilling down in that request shows this:

![image](https://user-images.githubusercontent.com/3605786/219370508-bf447bc9-dd59-470b-bd0b-068465f571d1.png)

So, the additional traces and custom dependency are not correlated to this request.  However, they're logged as I can find them via the 'transaction search' in app insights:

![image](https://user-images.githubusercontent.com/3605786/219370809-525fe088-5f19-429a-add5-e965a31f9cba.png)


@pauwelz can you confirm that you see the same behavior ?

**Comment by stijnmoreels** [comment]
That's different from my experience, and I also used a fresh template with only configuring Application Insights. 🤔

**Comment by fgheysels** [comment]
It's just the code that is provided by the Arcus Http Function template.

```csharp
 public static async Task Main(string[] args)
        {
            Log.Logger = new LoggerConfiguration()
                .MinimumLevel.Debug()
                .WriteTo.Console()
                .CreateBootstrapLogger();

            try
            {
                IHost host = CreateHostBuilder(args).Build();

                await ConfigureSerilogAsync(host);
                await host.RunAsync();
            }
            catch (Exception exception)
            {
                Log.Fatal(exception, "Host terminated unexpectedly");
            }
            finally
            {
                Log.CloseAndFlush();
            }
        }
```

```csharp
private static async Task ConfigureSerilogAsync(IHost app)
    {
        var secretProvider = app.Services.GetRequiredService<ISecretProvider>();
        string connectionString = await secretProvider.GetRawSecretAsync("APPLICATIONINSIGHTS_CONNECTION_STRING");

        var reloadLogger = (ReloadableLogger)Log.Logger;
        reloadLogger.Reload(config =>
        {
            
            config.MinimumLevel.Information()
                      .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
                      .Enrich.FromLogContext()
                      .Enrich.WithComponentName(app.Services)
                      .Enrich.WithVersion(app.Services)
                      .WriteTo.Console();

            if (!string.IsNullOrWhiteSpace(connectionString))
            {
                config.WriteTo.AzureApplicationInsightsWithConnectionString(app.Services, connectionString);
            }

            return config;
        });
    }
```
